src/query/query_executor.c

```diff
@@ -2248,6 +2249,11 @@ qexec_clear_xasl (THREAD_ENTRY * thread_p, xasl_node * xasl, bool is_final)
       pg_cnt += qexec_clear_agg_orderby_const_list (thread_p, xasl, is_final);
     }
 
+  if (xasl->sq_cache != NULL)
```

Comment:
Have you tested whether the `sq_cache_destroy` function should not be executed depending on `is_final`?

Reply:
We can adjust the extent of clearing based on `is_final`. However, currently, there doesn't seem to be any part inside the `sq_cache_destroy` that needs to be cleared dynamically based on whether `is_final` is true or not.